engine/crates/parser-sdl/src/federation.rs

```diff
@@ -61,20 +61,62 @@ pub enum EntityCachingConfig {
     Disabled,
     Enabled {
         ttl: Option<Duration>,
+        storage: EntityCacheStorage,
     },
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Default)]
+pub enum EntityCacheStorage {
+    #[default]
+    Memory,
+    Redis(RedisConfig),
+}
+
 impl From<gateway_config::EntityCachingConfig> for EntityCachingConfig {
     fn from(config: gateway_config::EntityCachingConfig) -> Self {
         match (config.enabled, config.ttl) {
             (Some(false), _) => EntityCachingConfig::Disabled,
-            (Some(true), ttl) => EntityCachingConfig::Enabled { ttl },
-            (_, Some(ttl)) => EntityCachingConfig::Enabled { ttl: Some(ttl) },
+            (Some(true), ttl) => EntityCachingConfig::Enabled {
+                ttl,
+                storage: entity_cache_storage(config.storage, config.redis),
+            },
+            (_, Some(ttl)) => EntityCachingConfig::Enabled {
+                ttl: Some(ttl),
+                storage: entity_cache_storage(config.storage, config.redis),
+            },
             _ => EntityCachingConfig::Disabled,
         }
     }
 }
 
+fn entity_cache_storage(
+    storage: gateway_config::EntityCachingStorage,
+    redis: Option<gateway_config::EntityCachingRedisConfig>,
+) -> EntityCacheStorage {
+    match storage {
+        gateway_config::EntityCachingStorage::Memory => EntityCacheStorage::Memory,
+        gateway_config::EntityCachingStorage::Redis => EntityCacheStorage::Redis(redis.unwrap_or_default().into()),
```

Comment:
:bug: **Bug**
The function `entity_cache_storage` does not handle the case where `redis` is `None` for the Redis storage variant. This could lead to a panic if `redis.unwrap_or_default()` is called when `redis` is `None`. You should handle this case more gracefully.

```suggestion
EntityCacheStorage::Redis(redis.unwrap_or_else(|| gateway_config::EntityCachingRedisConfig::default().into()))
```